v2/pkg/utils/ratelimit/ratelimit.go

```diff
@@ -0,0 +1,62 @@
+package ratelimit
+
+import (
+	"math"
+	"time"
+)
+
+// Limiter allows a burst of request during the defined duration
+type Limiter struct {
```

Comment:
wouldn't it be better to create a new library and implement this code there since the ratelimiter is used by all our tools and we would want them to have similar behaviours

Reply:
I think it's a good idea. What do you think about doing this within a follow-up ticket and implementing more tests, writing docs, and usage examples?

Reply:
+1, we can create a ticket for this, i'll create one in team-backlogs